runtime/util/hshelp.c

```diff
@@ -1701,6 +1701,72 @@ fixJNIRefs(J9VMThread * currentThread, J9HashTable * classPairs, BOOLEAN fastHCR
 }
 
 
+void
+fixMemberNames(J9JavaVM * vm, J9HashTable * classHashTable)
+{
+	PORT_ACCESS_FROM_JAVAVM(vm);
+
+	/* iterate over all objects fixing up vmtarget refs if object is a MemberName */
+	vm->memoryManagerFunctions->j9mm_iterate_all_objects(vm, PORTLIB, 0, fixMemberNamesObjectIteratorCallback, classHashTable);
+}
+
+static jvmtiIterationControl
+fixMemberNamesObjectIteratorCallback(J9JavaVM *vm, J9MM_IterateObjectDescriptor *objectDesc, void *userData)
+{
+	J9HashTable *classHashTable = userData;
+	j9object_t object = objectDesc->object;
+	J9Class *clazz = J9OBJECT_CLAZZ_VM(vm, object);
+
+	if (NULL != classHashTable) {
+		if (clazz == J9VMJAVALANGINVOKEMEMBERNAME(vm) && NULL != J9OBJECT_U64_LOAD(vm->currentThread, object, vm->vmindexOffset)) {
+			j9object_t vmindex = J9OBJECT_U64_LOAD(vm->currentThread, object, vm->vmindexOffset);
```

Comment:
vmindex is not a `j9object_t`, it is either a `J9JNIFieldID*` or `J9JNIMethodID*`. The defined type is U_64, which you can later cast to pointer explicitly